include/arch/arm/arch/object/vcpu.h

```diff
@@ -78,11 +78,11 @@ struct vcpu {
     word_t regs[seL4_VCPUReg_Num];
     bool_t vppi_masked[n_VPPIEventIRQ];
 #ifdef CONFIG_VTIMER_UPDATE_VOFFSET
-    word_t vcpu_padding;
+    /* word_t vcpu_padding; */
     /* vTimer is 8-bytes wide and has the same 8-byte alignment requirement.
      * If the sum of n_VPPIEventIRQ and seL4_VCPUReg_Num is even, we do not need
-     * extra padding. If the sum is odd we do. It currently is odd, so the extra
-     * padding above is necessary for the struct to remain packed on 32 bit
+     * extra padding. If the sum is odd we do. It currently is even, so the extra
+     * padding above is unnecessary for the struct to remain packed on 32 bit
```

Comment:
Good to add a compile_assert to check for this.

Also, the padding is only necessary for 32-bit, while it was always added?